### PR TITLE
Delete all tmp-drydock branches

### PR DESCRIPTION
**Pull request by goddenrich**
Use a regex to delete all tmp-drydock branches. The git delete might have failed previously so we clean all branches in the working copy including ones that might have been created in a previous lease. Also moving branch deletions after helpful error creations so we are less likely to see unhelpful cryptic error messages. If the merge is fully succesful we catch and continue on exception when cleaning up after ourselves by deleting a branch - The next time we will catch this with the branch deletion regex.

### CODE REVIEW

File: src/applications/drydock/blueprint/DrydockWorkingCopyBlueprintImplementation.php [hunk 1]
```diff
@@ -7,7 +7,7 @@ final class DrydockWorkingCopyBlueprintImplementation
   const PHASE_REMOTEFETCH = 'blueprint.workingcopy.fetch.remote';
   const PHASE_MERGEFETCH = 'blueprint.workingcopy.fetch.staging';
   // TM CHANGES BEGIN
-  const PHASE_MERGEREBASE = 'blueprint.workingcopy.rebase';
```
Comment (goddenrich) [inline src/applications/drydock/blueprint/DrydockWorkingCopyBlueprintImplementation.php:10]: rebase really wasn't a phase it was a path towards a successful sqash merge so removing it

File: src/applications/drydock/blueprint/DrydockWorkingCopyBlueprintImplementation.php [hunk 2]
```diff
@@ -569,14 +571,18 @@ private function applyMerge(
         $lease->setAttribute('workingcopy.vcs.error', $error->toDictionary());
         throw $ex;
       }
+      $interface->execx(
+        'git branch -D `git branch | grep -E "%s-.*"`',
+        self::GIT_BRANCH_PREFIX);
+      return;
```
Comment (goddenrich) [inline src/applications/drydock/blueprint/DrydockWorkingCopyBlueprintImplementation.php]: I realised we weren't deleting tmp-branches on success in the fallback case where we use the original behaviour.

File: src/applications/drydock/blueprint/DrydockWorkingCopyBlueprintImplementation.php [hunk 3]
```diff
@@ -7,7 +7,7 @@ final class DrydockWorkingCopyBlueprintImplementation
   const PHASE_REMOTEFETCH = 'blueprint.workingcopy.fetch.remote';
   const PHASE_MERGEFETCH = 'blueprint.workingcopy.fetch.staging';
   // TM CHANGES BEGIN
-  const PHASE_MERGEREBASE = 'blueprint.workingcopy.rebase';
+  const TMP_BRANCH_PREFIX = 'tmp-drydock';
```
Comment (Tatskaari) [inline src/applications/drydock/blueprint/DrydockWorkingCopyBlueprintImplementation.php:10]: Is this used anywhere?

Comment (goddenrich) [reply src/applications/drydock/blueprint/DrydockWorkingCopyBlueprintImplementation.php:10]: oops good spot I renamed it from GIT_BRANCH_PREFIX... I'll change it everywhere else

**Review by Tatskaari — APPROVED** [review]

**Pull request merged**